components/universal-widget/src/renderers/search-renderer/src/SearchRenderer.jsx

```diff
@@ -0,0 +1,421 @@
+            <Paper
+                square
+                style={{
+                    width: popperNode ?
+                        popperNode.clientWidth :
+                        document.getElementById(props.selectProps.id).clientWidth}}
+                {...props.innerProps}>
+                {props.children}
+            </Paper>
+        </Popper>
+    );
+};
+
+const components = {
+    Option,
+    Control,
+    NoOptionsMessage,
+    Placeholder,
+    SingleValue,
+    MultiValue,
+    ValueContainer,
+    Menu,
+};
+
+export default class SearchRenderer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            id: this.props.id,
+            width: this.props.width,
+            height: this.props.height,
+            columnIndex: this.props.metadata.names.indexOf(this.props.config.charts[0].column),
+            options: [],
+            availableOptions: [],
+            selectedOptions: null,
+            faultyProviderConf: false
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleDataReceived = this.handleDataReceived.bind(this);
+        this.naturalSort = this.naturalSort.bind(this);
+        this.publish = this.publish.bind(this);
+        this.updateStyleColor = this.updateStyleColor.bind(this);
+        this.updateTextBoxColor = this.updateTextBoxColor.bind(this);
+        this.setupPopperAnchorID = this.setupPopperAnchorID.bind(this);
+    }
+
+    componentWillReceiveProps(props) {
+        this.handleDataReceived(props.data);
+    }
+
+    /**
+     * Publish selection
+     * */
+    publish() {
+        if(this.props.config.widgetOutputConfigs
+            && this.state.selectedOptions) {
+            const { onClick } = this.props;
+            const data = {};
+            const selectedOptions = [];
+            this.state.selectedOptions.map(selection => { selectedOptions.push(selection.value) });
+            data[this.props.config.widgetOutputConfigs[0].publishingValue] = selectedOptions;
+            return onClick && onClick(data);
+        }
+    }
+
+    /**
+     * Create select options from data received
+     * */
+    handleDataReceived(data) {
+        let options = [], availableOptions;
+
+        if(this.state.options.length > 0) {
+            options = this.state.options;
+        }
+        data.forEach(dataUnit => {
+            const data = dataUnit[this.state.columnIndex];
+            if(options.indexOf(data) === -1) {
+                options.push(data);
+            }
+        });
+        options.sort(this.naturalSort);
+
+        availableOptions = options.map(option => ({
+            value: option,
+            label: option,
+            disabled: false
+        }));
+
+        this.setState({
+            options: options,
+            availableOptions: availableOptions,
+        });
+    }
+
+    /**
+     * Sort the options in asc order
+     * */
+    naturalSort(a, b) {
```

Comment:
Not required to write a sorting algo. Do as suggested in code review.

Reply:
This is the comparator function passed to the sort method, not a sort function.

Reply:
Code review suggestion was to remove sort functions. This is the comparator function passed to the sort method, not a sort function.

Reply:
Name of his function is confusing. Lets rename it to something like `naturalComparator`

Reply:
@ruwiniwj Can you explain what are the possible values for `a` and `b`?

Reply:
Renamed the function. 
a and b can be any 2 alphanumeric values, for example it could be 2 timestamps, 2 names etc